packages/fetchai/skills/tac_control/dialogues.py

```diff
@@ -143,36 +124,45 @@ def __init__(self, **kwargs) -> None:
         :return: None
         """
         Model.__init__(self, **kwargs)
-        BaseTacDialogues.__init__(self, self.context.agent_address)
-        self._dialogue_by_address = {}  # type: Dict[str, Dialogue]
-
-    @property
-    def dialogue_by_address(self) -> Dict[str, Dialogue]:
-        """Get the dialogue by address."""
-        return self._dialogue_by_address
-
-    @staticmethod
-    def role_from_first_message(message: Message) -> Dialogue.Role:
-        """Infer the role of the agent from an incoming/outgoing first message
-
-        :param message: an incoming/outgoing first message
-        :return: The role of the agent
-        """
-        return TacDialogue.Role.CONTROLLER
 
-    def create_dialogue(
-        self, dialogue_label: DialogueLabel, role: Dialogue.Role,
-    ) -> TacDialogue:
-        """
-        Create an instance of fipa dialogue.
-
-        :param dialogue_label: the identifier of the dialogue
-        :param role: the role of the agent this dialogue is maintained for
-
-        :return: the created dialogue
-        """
-        dialogue = TacDialogue(
-            dialogue_label=dialogue_label, agent_address=self.agent_address, role=role
+        def role_from_first_message(
+            message: Message, receiver_address: Address
+        ) -> Dialogue.Role:
+            """Infer the role of the agent from an incoming/outgoing first message
+
+            :param message: an incoming/outgoing first message
+            :param receiver_address: the address of the receiving agent
+            :return: The role of the agent
+            """
+            return TacDialogue.Role.CONTROLLER
+
+        BaseTacDialogues.__init__(
+            self,
+            agent_address=self.context.agent_address,
+            role_from_first_message=role_from_first_message,
         )
-        self._dialogue_by_address[dialogue_label.dialogue_opponent_addr] = dialogue
-        return dialogue
+        self._dialogue_by_address = {}  # type: Dict[str, Dialogue]
+
+    # ToDo replace the dialogue_by_address functionality if required with a method that dynamically
```

Comment:
Note here. The deprecated `create` method updated an auxiliary `dict` as well as creating a dialogue, which is the only place this method does something more than creating a dialogue. 

Moving to the new approach means this functionality is lost. 

If `dialogue_by_address`'s functionality cannot be achieved using standard fields and methods I suggest creating a method that upon accessing it dynamically goes through all dialogues and constructs the auxiliary dict. 

This is used 3 times in the behaviour and handler classes.

Reply:
This functionality needs to be added again. I don't see why we cannot simply store it as before..